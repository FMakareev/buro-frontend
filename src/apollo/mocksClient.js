import faker from 'faker';
import setupClient from './helpers/apolloClientMock';
import schema from './schema.graphqls';
import { userDocumentList } from './graphql/query/userDocumentList';
import { userList } from './graphql/query/userList';
import { userItem } from './graphql/query/userItem';
import { notificationList } from './graphql/query/notificationList';
import { ROLE_BANK, ROLE_BUREAU, ROLE_CLIENT } from '../shared/roles';

const defaultMocks = {
  Query: () => ({
    userList,
    userItem,
    userEmailItem: (query, { email }) => {
      console.log(query, email);
      switch (email) {
        case 'client@test.com': {
          return {
            ...userItem(),
            email: 'client@test.com',
            role: ROLE_CLIENT,
          };
        }
        case 'bank@test.com': {
          return {
            ...userItem(),
            email: 'bank@test.com',
            role: ROLE_BANK,
          };
        }
        case 'bureau@test.com': {
          return {
            ...userItem(),
            email: 'bureau@test.com',
            role: ROLE_BUREAU,
          };
        }
        default: {
          throw Error(
            JSON.stringify({
              userEmailItem: null,
              errors: [
                {
                  message: 'GraphQL error: user not found',
                },
              ],
            }),
          );
        }
      }
    },
    userDocumentList,
    notificationList,
  }),
  Mutation: () => ({
    /**
     * @params {func} mutation - этот же запрос
     * @params {object} props - аргументы которые были переданы
     * */
    createUser: (mutation, props) => props,
    updateUser: (mutation, props) => props,
    userResetPassword: (mutation, props) => props,
    createNotification: (mutation, props) =>
      // для имитации запроса к серверу с рандомной задержкой и результатом.
      new Promise((resolve, reject) => {
        setTimeout(() => {
          faker.random.number(1)
            ? resolve(JSON.stringify({ data: { createNotification: props } }))
            : reject(
                JSON.stringify({
                  errors: [
                    {
                      message: 'error!',
                    },
                  ],
                }),
              );
        }, faker.random.number(2000));
      }),
    updateNotification: (mutation, props) =>
      // для имитации запроса к серверу с рандомной задержкой и результатом.
      new Promise((resolve, reject) => {
        setTimeout(() => {
          faker.random.number(1)
            ? resolve(JSON.stringify({ data: { updateNotification: props } }))
            : reject(
                JSON.stringify({
                  errors: [
                    {
                      message: 'error!',
                    },
                  ],
                }),
              );
        }, faker.random.number(2000));
      }),
    userPasswordRecovery: (mutation, props) =>
      // throw Error(JSON.stringify({
      //   userPasswordRecovery: null,
      //   errors: [
      //     {
      //       message:'Error!'
      //     }
      //   ]
      // }))
      props,
  }),
};

export const mocksClient = setupClient(defaultMocks, schema)();

export default mocksClient;
