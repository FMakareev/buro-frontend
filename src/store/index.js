/* global isBrowser */
import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';
import { rootReducer } from './reducers/index';

const createStoreMiddleware = applyMiddleware(thunkMiddleware)(createStore);
const initialState = isBrowser ? window.PRELOADED_REDUX_STATE : {};
initialState.user = isBrowser ? JSON.parse(localStorage.getItem('user')) : null;

const Store = createStoreMiddleware(rootReducer, initialState, composeWithDevTools());

export { Store };

export default Store;
