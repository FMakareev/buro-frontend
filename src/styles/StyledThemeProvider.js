import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { TextVariant } from './variants/textVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#093971',
  color2: '#EFF9FC',
  color3: '#E7EDF8',
  color4: '#61C3E2',
  color5: '#A2A2A2',
  color6: '#0F60BD',
  color7: '#072C57',
  color8: '#61C3E2',
  color9: '#EB5757',
  color10: '#E7EDF8',
  color11: '',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  14,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['576px', '768px', '992px', '1200px'];

const boxShadow = ['4px 8px 16px rgba(28, 65, 84, 0.08)'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        text: TextVariant,
      },
      fontFamily: {
        regular: 'Montserrat Regular',
        medium: 'Montserrat Medium',
        bold: 'Montserrat Bold',
      },
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
