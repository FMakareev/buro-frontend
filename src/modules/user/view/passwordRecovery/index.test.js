import React from 'react';
import renderer from 'react-test-renderer';
import {ApolloProvider} from 'react-apollo';
import mocksClient from "../../../../apollo/mocksClient";

import {MemoryRouter} from 'react-router-dom';
import {PasswordRecovery} from './index';
import {StyledThemeProvider} from "../../../../styles/StyledThemeProvider";
import {Provider as ProviderRedux} from 'react-redux';
import {Store} from '../../../../store';


test('PasswordRecovery: без параметров адресной строки', () => {
  const output = renderer.create(<StyledThemeProvider>
    <ProviderRedux store={Store}>
      <MemoryRouter>
        <ApolloProvider client={mocksClient}>
          <PasswordRecovery/>
        </ApolloProvider>
      </MemoryRouter>
    </ProviderRedux>
  </StyledThemeProvider>);
  expect(output).toMatchSnapshot();
});

test('PasswordRecovery: рендер с параметрами строки', () => {
  const output = renderer.create(<StyledThemeProvider>
    <ProviderRedux store={Store}>
      <MemoryRouter
        initialEntries={["/password-recovery"]}
        initialIndex={0}
      >
        <ApolloProvider client={mocksClient}>
          <PasswordRecovery match={{
            params: {
              token: 'aasc3faw3wesegfw5gdefzsegyd'
            }
          }}/>
        </ApolloProvider>
      </MemoryRouter>
    </ProviderRedux>
  </StyledThemeProvider>);
  expect(output).toMatchSnapshot();
});

