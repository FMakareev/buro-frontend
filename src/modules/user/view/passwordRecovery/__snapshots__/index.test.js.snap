// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordRecovery: без параметров адресной строки 1`] = `
<div
  className="sc-bxivhb dUiVDW"
/>
`;

exports[`PasswordRecovery: рендер с параметрами строки 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <div
    className="sc-1jxwge6-0 sc-14k9gvj-0 sc-10559ym-0 iCJxjZ"
  >
    <div
      className="sc-1jxwge6-0 jHDLGC"
    >
      <div
        className="sc-1jxwge6-0 wterhv-0 eKGYIJ"
      >
        <div
          className="sc-1jxwge6-0 gHyFgL"
        >
          <svg
            fill="none"
            height="2.5625em"
            viewBox="0 0 36 41"
            width="2.25em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 40.0241C17.0435 40.0241 16.087 39.762 15.2174 39.2815L2.78261 32.0739C1.04348 31.0692 0 29.2345 0 27.2251V12.8099C0 10.8005 1.08695 8.96584 2.78261 7.96114L15.2174 0.753524C16.9565 -0.251175 19.087 -0.251175 20.7826 0.753524L33.2174 7.96114C34.9565 8.96584 36 10.8005 36 12.8099V27.2251C36 29.2345 34.913 31.0692 33.2174 32.0739L20.7826 39.2815C19.913 39.762 18.9565 40.0241 18 40.0241ZM18 0.884573C17.1739 0.884573 16.3913 1.10299 15.6522 1.49613L3.21739 8.74743C1.78261 9.5774 0.869565 11.15 0.869565 12.8099V27.2251C0.869565 28.8851 1.78261 30.4577 3.21739 31.2876L15.6522 38.4952C17.087 39.3252 18.8696 39.3252 20.3478 38.4952L32.7826 31.2876C34.2174 30.4577 35.1304 28.8851 35.1304 27.2251V12.8099C35.1304 11.15 34.2174 9.5774 32.7826 8.74743L20.3478 1.53981C19.6087 1.10299 18.7826 0.884573 18 0.884573Z"
              fill="url(#paint0_linear)"
            />
            <path
              d="M18.1304 10.6694C18.5217 10.6694 18.8695 10.8004 19.2173 10.9752L25.1304 14.4261C25.5652 14.6882 25.8695 15.0376 26.0869 15.4308L20.1304 4.29174C18.826 3.50545 17.1739 3.50545 15.8695 4.29174L5.52167 10.2763L18.1304 10.6694Z"
              fill="url(#paint1_linear)"
            />
            <path
              d="M10.0427 15.2124C10.2601 14.9066 10.5209 14.6008 10.8688 14.4261L16.7818 10.9752C17.2166 10.7131 17.6514 10.6257 18.1296 10.6694L5.52094 10.2762C4.21659 11.0625 3.3905 12.4604 3.3905 13.9893V25.9583L10.0427 15.2124Z"
              fill="url(#paint2_linear)"
            />
            <path
              d="M9.86946 24.5605C9.69554 24.211 9.6086 23.8616 9.6086 23.4684V16.6103C9.6086 16.1298 9.73903 15.6492 9.9999 15.2561L3.34772 26.0457C3.34772 27.5746 4.17382 28.9724 5.47816 29.7587L15.826 35.7432L9.86946 24.5605Z"
              fill="url(#paint3_linear)"
            />
            <path
              d="M17.8695 29.3218C17.4782 29.3218 17.1304 29.1908 16.7825 29.016L10.8695 25.5651C10.4347 25.303 10.1304 24.9536 9.91296 24.5604L15.8695 35.6995C17.1738 36.4858 18.826 36.4858 20.1304 35.6995L30.4782 29.715L17.8695 29.3218Z"
              fill="url(#paint4_linear)"
            />
            <path
              d="M25.9565 24.7789C25.7391 25.0846 25.4783 25.3904 25.1304 25.5651L19.2174 29.0161C18.7826 29.2782 18.3478 29.3655 17.8696 29.3218L30.4783 29.715C31.7826 28.9287 32.6087 27.5309 32.6087 26.002V14.033L25.9565 24.7789Z"
              fill="url(#paint5_linear)"
            />
            <path
              d="M26.0869 15.4745C26.2608 15.8239 26.3478 16.1734 26.3478 16.5665V23.4247C26.3478 23.9052 26.2173 24.3857 25.9565 24.7789L32.6086 13.9893C32.6086 12.4604 31.7826 11.0625 30.4782 10.2763L20.1304 4.29175L26.0869 15.4745Z"
              fill="url(#paint6_linear)"
            />
            <defs>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="paint0_linear"
                x1="17.9888"
                x2="17.9888"
                y1="0.683894"
                y2="40.3346"
              >
                <stop
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.2048"
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.876"
                  stopColor="#134177"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="paint1_linear"
                x1="7.61302"
                x2="26.4297"
                y1="4.79977"
                y2="13.7152"
              >
                <stop
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.2048"
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.876"
                  stopColor="#134177"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="paint2_linear"
                x1="-0.314949"
                x2="16.8723"
                y1="21.4428"
                y2="9.66638"
              >
                <stop
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.2048"
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.876"
                  stopColor="#134177"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="paint3_linear"
                x1="10.0478"
                x2="8.38388"
                y1="36.643"
                y2="15.8088"
              >
                <stop
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.2048"
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.876"
                  stopColor="#134177"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="paint4_linear"
                x1="28.3647"
                x2="9.54796"
                y1="35.2057"
                y2="26.2903"
              >
                <stop
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.2048"
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.876"
                  stopColor="#134177"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="paint5_linear"
                x1="36.301"
                x2="19.1137"
                y1="18.5895"
                y2="30.3659"
              >
                <stop
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.2048"
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.876"
                  stopColor="#134177"
                />
              </linearGradient>
              <linearGradient
                gradientUnits="userSpaceOnUse"
                id="paint6_linear"
                x1="25.9113"
                x2="27.5753"
                y1="3.3837"
                y2="24.218"
              >
                <stop
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.2048"
                  stopColor="#61C3E2"
                />
                <stop
                  offset="0.876"
                  stopColor="#134177"
                />
              </linearGradient>
            </defs>
          </svg>
        </div>
        <h2
          className="sc-1jxwge6-0 sc-18ch56c-0 cTSlNq"
          color="color1"
          fontFamily="bold"
          fontSize="24px"
        >
          New password
        </h2>
      </div>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="sc-1jxwge6-0 wterhv-0 ipWEwI"
        width="100%"
      >
        <div
          className="sc-1jxwge6-0 fKFOsG"
          width="100%"
        >
          <div
            className="sc-1tlq1yi-0 dLPklW"
          >
            <div
              className="sc-1jxwge6-0 wterhv-0 cGlObv"
            >
              <div
                className="sc-1jxwge6-0 sc-1tlq1yi-1 bjrTHb"
                width="100%"
              >
                <input
                  aria-required={false}
                  className="sc-1tlq1yi-2 fRLrXW wsrc07-0 kZIRDH"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <div
                  className="sc-1tlq1yi-3 ktWGUq"
                >
                  <div
                    className="sc-1jxwge6-0 sc-18ch56c-0 kapYeH"
                    color="color1"
                    fill="inherit"
                    fontSize={11}
                  >
                    <svg
                      fill="inherit"
                      height="1em"
                      stroke="none"
                      viewBox="0 0 32 32"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.271 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M23.4829 13H9.01712C8.45629 13 8 13.4577 8 14.0202V17.1296C8 17.6921 8.45629 18.1498 9.01712 18.1498H18.0582C18.192 18.1498 18.3004 18.041 18.3004 17.9069C18.3004 17.7727 18.192 17.664 18.0582 17.664H9.01712C8.72335 17.664 8.48434 17.4242 8.48434 17.1296V14.0202C8.48434 13.7256 8.72335 13.4858 9.01712 13.4858H23.4829C23.7767 13.4858 24.0157 13.7256 24.0157 14.0202V16.3522C24.0157 16.4864 24.1241 16.5951 24.2578 16.5951C24.3916 16.5951 24.5 16.4864 24.5 16.3522V14.0202C24.5 13.4577 24.0437 13 23.4829 13Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M23.4666 16.9266V16.6114C23.4666 15.4773 22.5467 14.5547 21.4162 14.5547C20.2856 14.5547 19.3658 15.4773 19.3658 16.6114V16.9266C19.0658 17.0283 18.8491 17.313 18.8491 17.6478V20.2389C18.8491 20.6586 19.1895 21 19.6079 21H23.2244C23.6428 21 23.9832 20.6586 23.9832 20.2389V17.6478C23.9832 17.313 23.7665 17.0283 23.4666 16.9266ZM21.4162 15.0405C22.2797 15.0405 22.9822 15.7452 22.9822 16.6114V16.8867H19.8501V16.6114C19.8501 15.7452 20.5526 15.0405 21.4162 15.0405ZM23.4988 20.2389C23.4988 20.3907 23.3757 20.5142 23.2244 20.5142H19.6079C19.4566 20.5142 19.3335 20.3907 19.3335 20.2389V17.6478C19.3335 17.496 19.4566 17.3725 19.6079 17.3725H23.2244C23.3757 17.3725 23.4988 17.496 23.4988 17.6478V20.2389Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M11.9301 14.9351C11.8632 14.8189 11.7151 14.7791 11.5993 14.8462L11.0677 15.154V14.5383C11.0677 14.4042 10.9593 14.2954 10.8255 14.2954C10.6918 14.2954 10.5834 14.4042 10.5834 14.5383V15.154L10.0518 14.8462C9.93595 14.7791 9.78783 14.8189 9.72096 14.9351C9.65409 15.0513 9.69377 15.1998 9.8096 15.2669L10.3412 15.5748L9.8096 15.8826C9.69377 15.9497 9.65409 16.0983 9.72096 16.2144C9.76581 16.2924 9.84721 16.3359 9.93091 16.3359C9.97198 16.3359 10.0136 16.3254 10.0518 16.3033L10.5834 15.9955V16.6112C10.5834 16.7454 10.6918 16.8541 10.8255 16.8541C10.9593 16.8541 11.0677 16.7454 11.0677 16.6112V15.9955L11.5993 16.3033C11.6374 16.3254 11.6791 16.3359 11.7201 16.3359C11.8038 16.3359 11.8852 16.2924 11.9301 16.2144C11.997 16.0983 11.9573 15.9497 11.8415 15.8826L11.3099 15.5748L11.8415 15.2669C11.9573 15.1998 11.997 15.0513 11.9301 14.9351Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M15.0297 14.9351C14.9628 14.8189 14.8147 14.7791 14.6989 14.8462L14.1673 15.154V14.5383C14.1673 14.4042 14.0589 14.2954 13.9251 14.2954C13.7914 14.2954 13.683 14.4042 13.683 14.5383V15.154L13.1514 14.8462C13.0356 14.7791 12.8874 14.8189 12.8206 14.9351C12.7537 15.0513 12.7934 15.1998 12.9092 15.2669L13.4408 15.5748L12.9092 15.8826C12.7934 15.9497 12.7537 16.0983 12.8206 16.2144C12.8654 16.2924 12.9468 16.3359 13.0305 16.3359C13.0716 16.3359 13.1132 16.3254 13.1514 16.3033L13.683 15.9955V16.6112C13.683 16.7454 13.7914 16.8541 13.9251 16.8541C14.0589 16.8541 14.1673 16.7454 14.1673 16.6112V15.9955L14.6989 16.3033C14.737 16.3254 14.7787 16.3359 14.8198 16.3359C14.9035 16.3359 14.9849 16.2924 15.0297 16.2144C15.0966 16.0983 15.0569 15.9497 14.9411 15.8826L14.4095 15.5748L14.9411 15.2669C15.0569 15.1998 15.0966 15.0513 15.0297 14.9351Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M18.1298 14.9351C18.0629 14.8189 17.9148 14.7791 17.799 14.8462L17.2674 15.154V14.5383C17.2674 14.4042 17.159 14.2954 17.0252 14.2954C16.8915 14.2954 16.7831 14.4042 16.7831 14.5383V15.154L16.2515 14.8462C16.1357 14.7791 15.9875 14.8189 15.9207 14.9351C15.8538 15.0513 15.8935 15.1998 16.0093 15.2669L16.5409 15.5748L16.0093 15.8826C15.8935 15.9497 15.8538 16.0983 15.9207 16.2144C15.9655 16.2924 16.0469 16.3359 16.1306 16.3359C16.1717 16.3359 16.2133 16.3254 16.2515 16.3033L16.7831 15.9955V16.6112C16.7831 16.7454 16.8915 16.8541 17.0252 16.8541C17.159 16.8541 17.2674 16.7454 17.2674 16.6112V15.9955L17.799 16.3033C17.8371 16.3254 17.8788 16.3359 17.9199 16.3359C18.0035 16.3359 18.085 16.2924 18.1298 16.2144C18.1967 16.0983 18.157 15.9497 18.0412 15.8826L17.5096 15.5748L18.0412 15.2669C18.157 15.1998 18.1967 15.0513 18.1298 14.9351Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M21.416 17.9229C20.9976 17.9229 20.6572 18.2643 20.6572 18.684C20.6572 19.0188 20.8739 19.3035 21.1739 19.4052V19.7204C21.1739 19.8546 21.2823 19.9633 21.416 19.9633C21.5498 19.9633 21.6582 19.8546 21.6582 19.7204V19.4052C21.9582 19.3035 22.1748 19.0188 22.1748 18.684C22.1748 18.2643 21.8344 17.9229 21.416 17.9229ZM21.416 18.9593C21.2647 18.9593 21.1416 18.8358 21.1416 18.684C21.1416 18.5322 21.2647 18.4087 21.416 18.4087C21.5674 18.4087 21.6905 18.5322 21.6905 18.684C21.6905 18.8358 21.5674 18.9593 21.416 18.9593Z"
                        fill="inherit"
                        stroke="none"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-1jxwge6-0 fKFOsG"
          width="100%"
        >
          <div
            className="sc-1tlq1yi-0 dLPklW"
          >
            <div
              className="sc-1jxwge6-0 wterhv-0 cGlObv"
            >
              <div
                className="sc-1jxwge6-0 sc-1tlq1yi-1 bjrTHb"
                width="100%"
              >
                <input
                  aria-required={false}
                  className="sc-1tlq1yi-2 fRLrXW wsrc07-0 kZIRDH"
                  name="confirmPassword"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Retype password"
                  type="password"
                  value=""
                />
                <div
                  className="sc-1tlq1yi-3 ktWGUq"
                >
                  <div
                    className="sc-1jxwge6-0 sc-18ch56c-0 kapYeH"
                    color="color1"
                    fill="inherit"
                    fontSize={11}
                  >
                    <svg
                      fill="inherit"
                      height="1em"
                      stroke="none"
                      viewBox="0 0 32 32"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.271 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M23.4829 13H9.01712C8.45629 13 8 13.4577 8 14.0202V17.1296C8 17.6921 8.45629 18.1498 9.01712 18.1498H18.0582C18.192 18.1498 18.3004 18.041 18.3004 17.9069C18.3004 17.7727 18.192 17.664 18.0582 17.664H9.01712C8.72335 17.664 8.48434 17.4242 8.48434 17.1296V14.0202C8.48434 13.7256 8.72335 13.4858 9.01712 13.4858H23.4829C23.7767 13.4858 24.0157 13.7256 24.0157 14.0202V16.3522C24.0157 16.4864 24.1241 16.5951 24.2578 16.5951C24.3916 16.5951 24.5 16.4864 24.5 16.3522V14.0202C24.5 13.4577 24.0437 13 23.4829 13Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M23.4666 16.9266V16.6114C23.4666 15.4773 22.5467 14.5547 21.4162 14.5547C20.2856 14.5547 19.3658 15.4773 19.3658 16.6114V16.9266C19.0658 17.0283 18.8491 17.313 18.8491 17.6478V20.2389C18.8491 20.6586 19.1895 21 19.6079 21H23.2244C23.6428 21 23.9832 20.6586 23.9832 20.2389V17.6478C23.9832 17.313 23.7665 17.0283 23.4666 16.9266ZM21.4162 15.0405C22.2797 15.0405 22.9822 15.7452 22.9822 16.6114V16.8867H19.8501V16.6114C19.8501 15.7452 20.5526 15.0405 21.4162 15.0405ZM23.4988 20.2389C23.4988 20.3907 23.3757 20.5142 23.2244 20.5142H19.6079C19.4566 20.5142 19.3335 20.3907 19.3335 20.2389V17.6478C19.3335 17.496 19.4566 17.3725 19.6079 17.3725H23.2244C23.3757 17.3725 23.4988 17.496 23.4988 17.6478V20.2389Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M11.9301 14.9351C11.8632 14.8189 11.7151 14.7791 11.5993 14.8462L11.0677 15.154V14.5383C11.0677 14.4042 10.9593 14.2954 10.8255 14.2954C10.6918 14.2954 10.5834 14.4042 10.5834 14.5383V15.154L10.0518 14.8462C9.93595 14.7791 9.78783 14.8189 9.72096 14.9351C9.65409 15.0513 9.69377 15.1998 9.8096 15.2669L10.3412 15.5748L9.8096 15.8826C9.69377 15.9497 9.65409 16.0983 9.72096 16.2144C9.76581 16.2924 9.84721 16.3359 9.93091 16.3359C9.97198 16.3359 10.0136 16.3254 10.0518 16.3033L10.5834 15.9955V16.6112C10.5834 16.7454 10.6918 16.8541 10.8255 16.8541C10.9593 16.8541 11.0677 16.7454 11.0677 16.6112V15.9955L11.5993 16.3033C11.6374 16.3254 11.6791 16.3359 11.7201 16.3359C11.8038 16.3359 11.8852 16.2924 11.9301 16.2144C11.997 16.0983 11.9573 15.9497 11.8415 15.8826L11.3099 15.5748L11.8415 15.2669C11.9573 15.1998 11.997 15.0513 11.9301 14.9351Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M15.0297 14.9351C14.9628 14.8189 14.8147 14.7791 14.6989 14.8462L14.1673 15.154V14.5383C14.1673 14.4042 14.0589 14.2954 13.9251 14.2954C13.7914 14.2954 13.683 14.4042 13.683 14.5383V15.154L13.1514 14.8462C13.0356 14.7791 12.8874 14.8189 12.8206 14.9351C12.7537 15.0513 12.7934 15.1998 12.9092 15.2669L13.4408 15.5748L12.9092 15.8826C12.7934 15.9497 12.7537 16.0983 12.8206 16.2144C12.8654 16.2924 12.9468 16.3359 13.0305 16.3359C13.0716 16.3359 13.1132 16.3254 13.1514 16.3033L13.683 15.9955V16.6112C13.683 16.7454 13.7914 16.8541 13.9251 16.8541C14.0589 16.8541 14.1673 16.7454 14.1673 16.6112V15.9955L14.6989 16.3033C14.737 16.3254 14.7787 16.3359 14.8198 16.3359C14.9035 16.3359 14.9849 16.2924 15.0297 16.2144C15.0966 16.0983 15.0569 15.9497 14.9411 15.8826L14.4095 15.5748L14.9411 15.2669C15.0569 15.1998 15.0966 15.0513 15.0297 14.9351Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M18.1298 14.9351C18.0629 14.8189 17.9148 14.7791 17.799 14.8462L17.2674 15.154V14.5383C17.2674 14.4042 17.159 14.2954 17.0252 14.2954C16.8915 14.2954 16.7831 14.4042 16.7831 14.5383V15.154L16.2515 14.8462C16.1357 14.7791 15.9875 14.8189 15.9207 14.9351C15.8538 15.0513 15.8935 15.1998 16.0093 15.2669L16.5409 15.5748L16.0093 15.8826C15.8935 15.9497 15.8538 16.0983 15.9207 16.2144C15.9655 16.2924 16.0469 16.3359 16.1306 16.3359C16.1717 16.3359 16.2133 16.3254 16.2515 16.3033L16.7831 15.9955V16.6112C16.7831 16.7454 16.8915 16.8541 17.0252 16.8541C17.159 16.8541 17.2674 16.7454 17.2674 16.6112V15.9955L17.799 16.3033C17.8371 16.3254 17.8788 16.3359 17.9199 16.3359C18.0035 16.3359 18.085 16.2924 18.1298 16.2144C18.1967 16.0983 18.157 15.9497 18.0412 15.8826L17.5096 15.5748L18.0412 15.2669C18.157 15.1998 18.1967 15.0513 18.1298 14.9351Z"
                        fill="inherit"
                        stroke="none"
                      />
                      <path
                        d="M21.416 17.9229C20.9976 17.9229 20.6572 18.2643 20.6572 18.684C20.6572 19.0188 20.8739 19.3035 21.1739 19.4052V19.7204C21.1739 19.8546 21.2823 19.9633 21.416 19.9633C21.5498 19.9633 21.6582 19.8546 21.6582 19.7204V19.4052C21.9582 19.3035 22.1748 19.0188 22.1748 18.684C22.1748 18.2643 21.8344 17.9229 21.416 17.9229ZM21.416 18.9593C21.2647 18.9593 21.1416 18.8358 21.1416 18.684C21.1416 18.5322 21.2647 18.4087 21.416 18.4087C21.5674 18.4087 21.6905 18.5322 21.6905 18.684C21.6905 18.8358 21.5674 18.9593 21.416 18.9593Z"
                        fill="inherit"
                        stroke="none"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-1jxwge6-0 hZdxoY"
          width="100%"
        >
          <button
            className="sc-72db93-0 bWXyHX mxzgdr-0 kHylNb"
            disabled={true}
            fontSize={6}
            size="medium"
            type="submit"
            width="100%"
          >
            <div
              className="sc-1jxwge6-0 wterhv-0 eZGOHC"
              width="100%"
            >
              <div />
              <div
                className="sc-1jxwge6-0 wterhv-0 irkFkV"
              >
                Create new password
              </div>
              <div
                className="sc-1jxwge6-0 wterhv-0 irkFkV"
              >
                <div
                  className="sc-1jxwge6-0 sc-18ch56c-0 hgJjUO"
                  color="color1"
                  fontSize={12}
                >
                  <svg
                    fill="inherit"
                    height="32px"
                    viewBox="0 0 40 41"
                    width="32px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M33.5631 20.9259C33.6718 20.8075 33.7261 20.6563 33.7261 20.5051C33.7261 20.3557 33.6729 20.2058 33.5661 20.0873L26.1245 11.8052C25.8975 11.5524 25.5118 11.5344 25.2627 11.7655C25.0137 11.9961 24.9973 12.3884 25.2237 12.6406L31.6978 19.8463L6.875 19.8463L6.875 21.0994L31.7417 21.0994L25.1688 28.3019C24.9393 28.5517 24.9539 28.944 25.2017 29.1771C25.4482 29.4094 25.8346 29.3946 26.0635 29.1436L33.5631 20.9259Z"
                      fill="inherit"
                    />
                    <path
                      d="M20 40.8095C31.0457 40.8095 40 31.7131 40 20.4921C40 9.27104 31.0457 0.174591 20 0.174591C8.95425 0.174591 0 9.27104 0 20.4921C0 31.7131 8.95425 40.8095 20 40.8095ZM20 39.5397C9.66119 39.5397 1.25 30.995 1.25 20.4921C1.25 9.98919 9.66119 1.44443 20 1.44443C30.3388 1.44443 38.75 9.98919 38.75 20.4921C38.75 30.995 30.3388 39.5397 20 39.5397Z"
                      fill="inherit"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
